source/develop/release-management/features/network/lldp.html.md

```diff
@@ -0,0 +1,277 @@
+---
+title: Link Layer Discovery Protocol (LLDP) Support
+category: feature
+authors: Dominik Holler
+wiki_category: Feature
+wiki_title: Link Layer Discovery Protocol (LLDP) Support
+wiki_revision_count: 1
+wiki_last_updated: 2016-06-27
+feature_name: Link Layer Discovery Protocol (LLDP) Support
+feature_modules: engine,network,vdsm,rest-api
+feature_status: In Development
+---
+
+# Link Layer Discovery Protocol (LLDP) Support
+
+## Summary
+
+The Link Layer Discovery Protocol (LLDP) [1] is a vendor-neutral networking
+protocol, used by network devices, especially switches, for advertising their
+identity, capabilities and management addresses on an IEEE 802 local area
+network (LAN).
+
+The feature will enable oVirt hosts to gather information from their networking
+interfaces and let oVirt engine use this information in a beneficial way.
+
+## Owner
+
+
+*   Name: [Dominik Holler](https://github.com/dominikholler)
+
+*   Email: <dholler@redhat.com>
+
+## Detailed Description
+
+To support the integration of oVirt in other software, the information gathered
+by LLDP will be provided by oVirt engin's REST-API.
+
+Some of the information, like the number of the switch port a network interface
+of the host is connected to, might be helpful for the user and will be shown in
+the graphical user interface.
+
+A third benefit of this feature is, that the information can be used by engine
+to configure the hosts in a smarter way. Engine could detect and avoid host
+networking configurations, which does not match the configuration of the switch
+port the relevant host's network interface is connected to.
+Example of relevant configuration options are:
+
+* The VLAN configuration of the host has to match the VLAN configuration of the
+  switch port. Other way roud: If a host should be connected to a given VLAN,
+  there might be only a single network interface of the host which is connected
+  to a switch port which allows the usage of this VLAN.
+
+* According to section "5. Switch Configuration" in [2], the bonding modes
+  balance-rr (0), balance-xor (2), broadcast (3), 802.3ad (4) may require
+  configuration on switch side.
+
+* The switch port might have constraints about the MTU.
+
+
+The host will not expose information by LLDP to the switch.
+
+## Prerequisites
+
+LLDP has to be enabled on the switch connected to the relevant oVirt host.
+
+## Limitations
+
+* Because the initial LLDP data are in a non-uniform format, they will be mapped
```

Comment:
this is not a limitation, but a challenge we must overcome.

LLDP data are  in a non-uniform format (each TLV has its own format). In order to be exposed via RESt, we would need to map them to a uniform format.